docs/hugo/content/design/ADR-2023-02-Helm-Chart-Size-Limitations.md

```diff
@@ -0,0 +1,108 @@
+
+I think that the best option for chart split is by RP/group. It ends up with a clean structure that matches SDKs. Other 
+possible splits seem to risk some charts becoming too large, or a confusing experience for customers. Imagine "what chart
+contains RoleAssignment again?"
+
+**Pros**:
+- It's clear which resource goes where, and even for the large RPs like `compute` and `networking` there should be plenty
+  of space in each chart to grow and add new resources/versions, at least in the near term.
+
+**Cons**:
+- There's no way with Helm out of the box to get an all-in-one installation, which is the current default now.
+  There are tools like [Helmfile](https://github.com/helmfile/helmfile) that could enable an all-in-one story.
+- There are a lot of RPs, which translates into a lot of Helm charts. The number of charts might get a bit overwhelming.
+- It's possible we run into this problem again with one of the huge RPs like `compute` or `networking` a few years down
+  the line if we keep adding new versions and new resources.
+
+### Avoiding Helm (for installing CRDs)
+
+Helm has pretty poor CRD support. In fact, we install CRDs via Helm by treating them like normal resources. We
+have to do this because otherwise Helm never upgrades them at all. We could just stop using Helm to install CRDs entirely.
+
+There are a few options for this:
+1. We have users install CRDs manually via `kubectl`.
+2. We update `asoctl` to support installing CRDs for users.
+3. We embed the CRDs into the operator pod and have the operator pod manage them on startup.
+
+For all of the above options, we would still maintain a Helm chart for the operator itself (the deployment, service account, etc).
+
+Of the above options, I believe embedding the CRDs into the operator pod is the best option. Here are some pros/cons for it.
+
+**Pros**:
+- Helm wasn't really getting us a lot for managing CRDs anyway, and embedding the CRDs into the operator itself
+  means that we can guarantee the right versions are installed for the corresponding operator version (unlike all
+  the chart-splitting options above)
+- We're able to continue to offer an easy-to-use Helm chart that provides the entire ASO feature set.
+- Because it's integrated into the operator we could add safeguards and capabilities to this "CRD reconciliation" 
+  for both deprecation or uninstall cases over and above what we could easily do in Helm.
+
+**Cons**:
+- CRD installation would _probably_ be a required pre-step and if it failed the operator would stop running, which means
+  existing resources wouldn't be reconciled until the deployment was rolled back. We could possibly mitigate this by making
+  CRD installation failure nonfatal, but that has other complications.
+- We would need to expose levers to the user (possibly via ConfigMap?) for the user to configure what CRDs get installed
```

Comment:
I see this as a pro and not a con? As with helm chart this would have been painful, however now with this change, we can manage CRDs and registers in the same place.

Reply:
Yeah I see where you are coming from. I've rephrased this slightly and moved it to Pro